ui/vic-ui-h5c/vic/src/vic-webapp/src/app/create-vch-wizard/security/security.component.spec.ts

```diff
@@ -0,0 +1,120 @@
+import {Observable} from 'rxjs/Observable';
+import {SecurityComponent} from './security.component';
+
+describe('SecurityComponent', () => {
+
+  let component: SecurityComponent;
+  let fixture: ComponentFixture<SecurityComponent>;
+  let service: CreateVchWizardService;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        ReactiveFormsModule,
+        HttpModule,
+        ClarityModule
+      ],
+      providers: [
+        {
+          provide: CreateVchWizardService,
+          useValue: {
+            getUserId() {
+              return Observable.of('userId');
+            },
+            getServerThumbprint() {
+              return Observable.of('serverThumbprint');
+            },
+            getVcHostname() {
+              return Observable.of('vcHostname');
+            }
+          }
+        }
+      ],
+      declarations: [
+        SecurityComponent
+      ]
+    });
+  });
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(SecurityComponent);
+    component = fixture.componentInstance;
+    component.onPageLoad();
+
+    service = fixture.debugElement.injector.get(CreateVchWizardService);
+    spyOn(service, 'getUserId').and.callThrough();
+    spyOn(service, 'getServerThumbprint').and.callThrough();
+    spyOn(service, 'getVcHostname').and.callThrough();
+  });
+
+  it('should be created', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should start with a valid form',  () => {
+    component.onCommit();
+    expect(component.form.valid).toBe(true);
+  });
+
+  it('should validate advanced fields defaults values', () => {
+    component.toggleAdvancedMode();
+    component.onCommit();
+    expect(component.form.valid).toBe(true);
+  });
+
+  it('should validate advanced fields changes', () => {
+    component.toggleAdvancedMode();
+
+    component.form.get('useTls').setValue(false);
+    expect(component.form.get('tlsCertPath').disabled).toBeTruthy();
+    expect(component.form.get('certificateKeySize').disabled).toBeTruthy();
+
+    component.form.get('useTls').setValue(true);
+    expect(component.form.get('tlsCertPath').enabled).toBeTruthy();
+    expect(component.form.get('certificateKeySize').enabled).toBeTruthy();
+
+    component.form.get('useWhitelistRegistry').setValue(false);
+    expect(component.form.get('whitelistRegistries').disabled).toBeTruthy();
+
+    component.form.get('useWhitelistRegistry').setValue(true);
+    expect(component.form.get('whitelistRegistries').enabled).toBeTruthy();
+  });
+
+  it('should add and remove client certificate entries', () => {
+    component.addNewFormArrayEntry('tlsCas');
+    expect(component.form.get('tlsCas')['controls'].length).toBe(2);
+    component.removeFormArrayEntry('tlsCas', 1);
+    expect(component.form.get('tlsCas')['controls'].length).toBe(1);
+    component.removeFormArrayEntry('tlsCas', 0);
+    expect(component.form.get('tlsCas')['controls'].length).toBe(1);
```

Comment:
This seems like expected length should be 0 here

Reply:
Not really. It does not remove the last one (only its contents) so the user can add new values. I'll add a comment.